xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp

```diff
@@ -1859,24 +1868,37 @@ int CAMLCodec::Decode(uint8_t *pData, size_t iSize, double dts, double pts)
 
   // keep hw buffered demux above 1 second
   if (GetTimeSize() < target_timesize)
+  {
+    m_bDeque = false;
     rtn |= VC_BUFFER;
+  }
+  else
+    m_bDeque = true;
 
-  // wait until we get a new frame or 25ms,
-  if (m_ptsQueue.size() == 0)
-    m_ready_event.WaitMSec(25);
-
-  if (m_ptsQueue.size() > 0)
   {
     CSingleLock lock(m_ptsQueueMutex);
-    m_cur_pts = m_ptsQueue.front();
-    m_ptsQueue.pop_front();
-    rtn |= VC_PICTURE;
+    if (!rtn && m_ptsQueue.size() == 0)
+    {
+      lock.Leave();
+      m_ready_event.WaitMSec(25);
+      lock.Enter();
+    }
+    if (m_ptsQueue.size() > 0)
+    {
+      m_cur_pts = m_ptsQueue.front();
+      m_ptsQueue.pop_front();
+      rtn |= VC_PICTURE;
+    }
   }
-/*
-  CLog::Log(LOGDEBUG, "CAMLCodec::Decode: "
-    "rtn(%d), m_cur_pictcnt(%lld), m_cur_pts(%f), lastpts(%f), GetTimeSize(%f), GetDataSize(%d)",
-    rtn, m_cur_pictcnt, (float)m_cur_pts/PTS_FREQ, (float)am_private->am_pkt.lastpts/PTS_FREQ, GetTimeSize(), GetDataSize());
-*/
+
+  if (g_advancedSettings.CanLogComponent(LOGVIDEO))
+    CLog::Log(LOGDEBUG, "CAMLCodec::Decode: ret: %d dts_in: %0.6f, pts_in: %0.6f, ptsOut:%0.6f"
+      ,rtn
+      ,static_cast<float>(dts)/DVD_TIME_BASE
+      ,static_cast<float>(pts)/DVD_TIME_BASE
+      ,static_cast<float>(m_cur_pts)/PTS_FREQ
+    );
```

Comment:
form ergonomic pov its easier at the beginning, because you can comment out things better, especially the last line. But Ok for me to change if you like it more